plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlJdbcConnectionCreation.java

```diff
@@ -0,0 +1,185 @@
+                {"INSERT INTO copy_of_nation SELECT * FROM nation", 15, Optional.empty()},
+                {"DELETE FROM copy_of_nation WHERE nationkey = 3", 4, Optional.empty()},
+                {"UPDATE copy_of_nation SET name = 'POLAND' WHERE nationkey = 1", 2, Optional.of("This connector does not support updates")},
+                {"MERGE INTO copy_of_nation n USING region r ON r.regionkey= n.regionkey WHEN MATCHED THEN DELETE", 2, Optional.of("This connector does not support merges")},
+                {"DROP TABLE copy_of_nation", 3, Optional.empty()},
+                {"SHOW SCHEMAS", 1, Optional.empty()},
+                {"SHOW TABLES", 2, Optional.empty()},
+                {"SHOW STATS FOR nation", 4, Optional.empty()},
+        };
+    }
+
+    private static DistributedQueryRunner createPostgreSqlQueryRunner(
+            TestingPostgreSqlServer server,
+            Iterable<TpchTable<?>> tables,
+            ConnectionCountingConnectionFactory connectionCountingConnectionFactory)
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = null;
+        try {
+            queryRunner = DistributedQueryRunner.builder(createSession())
+                    .build();
+
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+
+            Map<String, String> connectorProperties = new HashMap<>();
+            connectorProperties.putIfAbsent("connection-url", server.getJdbcUrl());
+            connectorProperties.putIfAbsent("connection-user", server.getUser());
+            connectorProperties.putIfAbsent("connection-password", server.getPassword());
+
+            queryRunner.installPlugin(new TestingPostgreSqlPlugin(connectionCountingConnectionFactory));
+            queryRunner.createCatalog("postgresql", "postgresql", connectorProperties);
+
+            copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, createSession(), tables);
+
+            return queryRunner;
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner);
+            throw e;
+        }
+    }
+
+    private static class TestingPostgreSqlPlugin
+            implements Plugin
+    {
+        private final ConnectionCountingConnectionFactory connectionCountingConnectionFactory;
+
+        private TestingPostgreSqlPlugin(ConnectionCountingConnectionFactory connectionCountingConnectionFactory)
+        {
+            this.connectionCountingConnectionFactory = requireNonNull(connectionCountingConnectionFactory, "connectionCountingConnectionFactory is null");
+        }
+
+        @Override
+        public Iterable<ConnectorFactory> getConnectorFactories()
+        {
+            return ImmutableList.of(new JdbcConnectorFactory(
+                    "postgresql",
+                    combine(
+                            new PostgreSqlClientModule(),
+                            new TestingPostgreSqlModule(connectionCountingConnectionFactory),
```

Comment:
Instead of creating a new `TestingPostgreSqlPlugin` can we combine these module and create a `JdbcPlugin` like this `new JdbcPlugin("jdbc", combine(new PostgreSqlClientModule(), new TestingPostgreSqlModule())`

Reply:
Good one!